io.asgardeo.java.oidc.sdk/src/main/java/io/asgardeo/java/oidc/sdk/HTTPSessionBasedOIDCProcessor.java

```diff
@@ -64,8 +65,13 @@ public void sendForLogin(HttpServletRequest request, HttpServletResponse respons
         HttpSession session = request.getSession();
         RequestContext requestContext = defaultOIDCManager.sendForLogin(request, response);
         if (request.getRequestURI() != null) {
-            String redirectPageURI = request.getRequestURI().substring(request.getContextPath().length() + 1);
-            requestContext.setParameter(SSOAgentConstants.REDIRECT_URI_KEY, redirectPageURI);
+            StringBuilder redirectPageURI=new StringBuilder();
+            redirectPageURI.append(request.getRequestURI().substring(request.getContextPath().length() + 1));
+            if (StringUtils.isNotBlank(request.getQueryString())) {
+                redirectPageURI.append("?");
+                redirectPageURI.append(request.getQueryString());
```

Comment:
```suggestion
                redirectPageURI.append("?").append(request.getQueryString());
```
let's combine these two

Reply:
Fixed